### ISSUE TITLE
detect.py fails without internet connection, always tries to load pre-trained model weights

### ISSUE THREAD

**Issue by r-blmnr** [ultralytics/yolov5#2799]
First of all, thanks for this great piece of software!

### Problem
Without internet connection, calling detect.py fails. It seems that even when a local copy of weights is available, the code still tries to load a set of weights from an online resource. 

Commenting this line prevents the error and the code runs as expected: 
https://github.com/ultralytics/yolov5/blob/e5d71223b83b9de2911a3d53685de6a20a2dc0f1/models/experimental.py#L117

However, as I am using the yolov5 version from PyPI, which is dynamically installed into a virtual environment, commenting is not a long-term option.

### Request
@glenn-jocher Would it be possible to include a switch, where the code only tries to load weights from an online resource if the weights are not found locally? This would also be in line with the comment at:
https://github.com/ultralytics/yolov5/blob/e5d71223b83b9de2911a3d53685de6a20a2dc0f1/utils/google_utils.py#L20

Thanks in advance!

**Comment by glenn-jocher** [comment]
@r-blmnr hi, thanks for the feature request! Models are already only searched for online if they are not found locally in L23 below. This feature has been in place for several months. Please git pull or git clone to use the latest code.

https://github.com/ultralytics/yolov5/blob/aff03be35a8f5c7fb7da8bfd2f26a93cde416fbc/utils/google_utils.py#L19-L53

**Comment by r-blmnr** [comment]
@glenn-jocher thanks for the quick reply. I have been using version 4.0.12 from PyPI, where line 25 from the code excerpt in your reply was still executed before checking whether the weight file exists. However, I just saw that this has been fixed in release 5.0.0 (issue #1890). 

While working on this I found another problem with my project structure:
My weight files were stored in a directory that contained uppercase characters. In L21 in the code excerpt above however, the file path is converted to lowercase, after which my model weights "did not exist".

Upgrading to version 5.0.0 and converting directory names to lower case solved my issue. Thanks again.
